test/server/test-run-test.js

```diff
@@ -0,0 +1,248 @@
+var expect          = require('chai').expect;
+var Promise         = require('pinkie');
+var noop            = require('lodash').noop;
+var TestRun         = require('../../lib/runner/test-run');
+var commands        = require('../../lib/runner/test-run/commands');
+var CLIENT_MESSAGES = require('../../lib/runner/client-messages');
+
+function wait (ms) {
+    return new Promise(function (resolve) {
+        setTimeout(resolve, ms);
+    });
+}
+
+function nextTick () {
+    return wait(0);
+}
+
+describe('TestRun', function () {
```

Comment:
Agree. But now this is the only way to be sure it works. We will change this with functional tests later